plugins/services/src/js/filters/TasksStatusFilter.js

```diff
@@ -0,0 +1,43 @@
+import DSLFilterTypes from "#SRC/js/constants/DSLFilterTypes";
+import DSLFilter from "#SRC/js/structs/DSLFilter";
+
+const LABEL = "is";
+
+const LABEL_TO_STATUS = {
+  active: "TASK_RUNNING",
+  completed: "TASK_COMPLETE"
+};
+
+/**
+ * This filter handles the `is:state` for tasks
+ */
+class TasksStatusFilter extends DSLFilter {
+  /**
+   * Handle all `is:XXXX` attribute filters that we can handle.
+   *
+   * @override
+   */
+  filterCanHandle(filterType, filterArguments) {
+    return (
+      filterType === DSLFilterTypes.ATTRIB &&
+      filterArguments.label === LABEL &&
+      LABEL_TO_STATUS[filterArguments.text.toLowerCase()] != null
+    );
+  }
+
+  /**
+   * Keep only tasks whose state matches the value of
+   * the `is` label
+   *
+   * @override
+   */
+  filterApply(resultset, filterType, filterArguments) {
```

Comment:
Why is filterType passed into when not being used?

Reply:
Although not used, we cannot change this as the callback that triggers this will pass the parameters in the exact order specified. If we remove it the filterArguments will be the filterType.